ui/jobs/components/brd.ts

```diff
@@ -57,20 +50,47 @@ export class BRDComponent extends BaseComponent {
     });
     this.repertoireTimer.toward = 'right';
     this.repertoireTimer.stylefill = 'fill';
-    // Only with-DoT-target you last attacked will trigger bars timer.
-    // So it work not well in multiple targets fight.
-    this.repertoireTracker = new DotTracker({ emitter: this.emitter, player: this.player });
-    this.repertoireTracker.onTick([
-      EffectId.Stormbite,
-      EffectId.Windbite,
-      EffectId.CausticBite,
-      EffectId.VenomousBite,
-    ], () => {
-      this.repertoireTimer.duration = kDoTTickInterval;
-    });
+
+    this.repertoireTracker5x = new DotTracker({ emitter: this.emitter, player: this.player });
+    if (this.is5x) {
+      // Only with-DoT-target you last attacked will trigger bars timer.
+      // So it work not well in multiple targets fight.
+      this.repertoireTracker5x.onTick([
+        EffectId.Stormbite,
+        EffectId.Windbite,
+        EffectId.CausticBite,
+        EffectId.VenomousBite,
+      ], () => {
+        this.repertoireTimer.duration = kDoTTickInterval;
+      });
+    } else {
+      this.repertoireTimer.loop = true;
+    }
+
     this.soulVoiceBox = this.bars.addResourceBox({
       classList: ['brd-color-soulvoice'],
     });
+
+    this.straightShotProc = this.bars.addProcBox({
+      id: 'brd-procs-straightshotready',
+      fgColor: 'brd-color-straightshotready',
+      threshold: 1000,
+    });
+    this.straightShotProc.bigatzero = false;
+  }
+
+  override onUseAbility(id: string): void {
+    switch (id) {
+      case kAbility.MagesBallad:
+      case kAbility.ArmysPaeon:
+      case kAbility.theWanderersMinuet:
+        // Seem EW BRD's repertoire always tick every 3s after song start
+        // 45s and 0s not included
+        // FIXME: stop timer when song is at last 3s or ended
+        if (!this.is5x)
+          this.repertoireTimer.duration = 3;
```

Comment:
Is it every 3 seconds, or is this the same as `kDoTTickInterval`? Is it tied to the start of the song, or is this like the mp ticker where it's based on server ticks?

Reply:
Every 3 seconds (at least I can't tell any mismatch in 45s)
Tied to the start of the song, Tick on 42 39 36 33 30 27 24 21 18 15 12 9 6 3

Reply:
Interesting!!

Reply:
Square normalized it heavily, for better or for worse, and we do indeed not get any benefit for running the song to zero.